### PR TITLE
Issue #978 - Fix application rollback to deployment without overrides

### LINKED ISSUE #978: Rollback to deployment without parameter overrides works incorrectly
Steps to reproduce:

1. Create any app and sync it
2. Override any parameter and sync again
3. Rollback to previous version ( version without parameter overrides )

Actual result:

Application is deployed with parameter overrides

Expected result:

Application is deployed **with out** parameter overrides

### PR DESCRIPTION
**Pull request by alexmt**
Closes #978

### CODE REVIEW

File: controller/sync_test.go [hunk 1]
```diff
@@ -5,13 +5,16 @@ import (
 	"sort"
 	"testing"
 
+	"github.com/argoproj/argo-cd/reposerver/repository"
```
Comment (dthomson25) [inline controller/sync_test.go]: Can you move this import into the internal project import section at line 22?

File: controller/state.go [hunk 2]
```diff
@@ -294,14 +294,17 @@ func (m *appStateManager) getRepo(repoURL string) *v1alpha1.Repository {
...
 }
 
-func (m *appStateManager) persistRevisionHistory(app *v1alpha1.Application, revision string, overrides *[]v1alpha1.ComponentParameter) error {
+func (m *appStateManager) persistRevisionHistory(app *v1alpha1.Application, revision string, overrides []v1alpha1.ComponentParameter) error {
 
 	var nextID int64 = 0
 	if len(app.Status.History) > 0 {
 		nextID = app.Status.History[len(app.Status.History)-1].ID + 1
 	}
+	if overrides == nil {
```
Comment (dthomson25) [inline controller/state.go]: if overrides isn't a pointer anymore, how can that variable be nil

Comment (dthomson25) [reply controller/state.go]: Oh it's an array 🤦‍♂️

File: controller/sync_test.go (hunk 1 above)
Comment (alexmt) [reply controller/sync_test.go]: Sure, done: https://github.com/argoproj/argo-cd/pull/979/commits/1a816900d6ab9469725cb40d9d5f969e08eef095

File: controller/state.go (hunk 2 above)
Comment (alexmt) [reply controller/state.go]: Yeah, https://github.com/argoproj/argo-cd/pull/979/files#diff-834370de48b922435d85e0231a6b8499R294 shows how it could happen

**Review by dthomson25 — APPROVED** [review]
LGTM!

**Pull request merged**